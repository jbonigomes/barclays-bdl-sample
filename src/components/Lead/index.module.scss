@import '../../index.scss';

.lead {
  font-size: 20px;
  color: $dark-blue;
  line-height: 32px;
  text-align: center;
  margin-bottom: 8px;
  font-family: 'ExpertSans-Light';

  @media (min-width: $breakpoint) {
    text-align: left;
  }
}
