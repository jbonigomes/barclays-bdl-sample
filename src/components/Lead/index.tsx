import React from 'react'
import styles from './index.module.scss'
import { dynamicHeader } from '../../helpers'

type LeadProps = {
  text: string,
  headerLevel?: number,
}

const Lead: React.SFC<LeadProps> = ({ text, headerLevel }: LeadProps) => {
  const DynamicHeader = dynamicHeader(headerLevel)

  return (
    <DynamicHeader className={`${styles.lead}`}>
      {text}
    </DynamicHeader>
  )
}

export default Lead
