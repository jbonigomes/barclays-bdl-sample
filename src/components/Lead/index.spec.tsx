import React from 'react'
import renderer from 'react-test-renderer'

import Lead from '.'

test('Match snapshot', () => {
  const props = {
    text: 'text',
  }

  const tree = renderer.create(<Lead text={props.text} />)

  expect(tree.toJSON()).toMatchSnapshot()
})
