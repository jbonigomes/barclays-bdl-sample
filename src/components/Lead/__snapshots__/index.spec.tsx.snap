// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
<h1
  className="lead"
>
  text
</h1>
`;
